crates/bevy_input/src/gamepad/settings.rs

```diff
@@ -0,0 +1,663 @@
+use crate::gamepad::{GamepadAxis, GamepadButton};
+use bevy_utils::HashMap;
+
+/// A setting for all [`Gamepad`](crate::gamepad::Gamepad)s.
+///
+/// ## Usage
+///
+/// It is used to create a `Bevy` resource that stores the settings of every [`GamepadButton`] and
+/// [`GamepadAxis`]. If no user defined [`ButtonSettings`], [`AxisSettings`], or [`ButtonAxisSettings`]
+/// are defined, the default settings of each are used as a fallback accordingly.
+///
+/// ## Access
+///
+/// To access the resource use one of the following:
+/// - Non-mutable access of the gamepad settings: `Res<GamepadSettings>`
+/// - Mutable access of the gamepad settings: `ResMut<GamepadSettings>`
+///
+/// ## Usage
+///
+/// The [`GamepadSettings`] are used inside of the [`gamepad_event_system`][crate::gamepad::gamepad_event_system],
+/// but are never written to inside of `Bevy`. To insert user defined settings it is required to mutably access
```

Comment:
We can use simple language here: e.g. "To modify these settings, mutate the corresponding resource."